N2A/src/gov/sandia/n2a/db/AppData.java

```diff
@@ -80,19 +84,33 @@ public class AppData
             state.set (primary, "Repos", "primary");
         }
 
-        List<MNode> modelContainers     = new ArrayList<MNode> ();
-        List<MNode> referenceContainers = new ArrayList<MNode> ();
+        Map<String, List<MNode>> containers = new HashMap<String, List<MNode>> ();
         for (String repoName : reposOrder)
         {
             MNode repo = repos.child (repoName);
             if (repo == null  ||  ! repo.getBoolean ("visible")  &&  ! repo.getBoolean ("editable")  &&  ! repoName.equals (primary)) continue;
             Path repoDir = reposDir.resolve (repoName);
-            modelContainers    .add (new MDir (repoName, repoDir.resolve ("models")));
-            referenceContainers.add (new MDir (repoName, repoDir.resolve ("references")));
+            try (DirectoryStream<Path> stream = Files.newDirectoryStream (repoDir))
+            {
+                for (Path subfolder : stream)
+                {
+                    String key = subfolder.getFileName ().toString ();
+                    if (key.startsWith (".")) continue;
+                    if (! Files.isDirectory (subfolder)) continue;
```

Comment:
If you put this test first, you can avoid doing the extra work of fetching key when it is unneeded. While this is basically irrelevant considering the speed of a modern processor, writing logically optimal code is good style.